marconi/src/Marconi/Index/ScriptTx.hs

```diff
@@ -275,21 +300,35 @@ instance Queryable ScriptTxHandle where
 
 instance Rewindable ScriptTxHandle where
   rewindStorage
-    :: SlotNo
+    :: ChainPoint
     -> ScriptTxHandle
     -> IO (Maybe ScriptTxHandle)
-  rewindStorage sn h@(ScriptTxHandle c _) = do
+  rewindStorage (ChainPoint sn   _) h@(ScriptTxHandle c _) = do
      SQL.execute c "DELETE FROM script_transactions WHERE slotNo > ?" (SQL.Only sn)
      pure $ Just h
+  rewindStorage ChainPointAtGenesis h@(ScriptTxHandle c _) = do
+     SQL.execute_ c "DELETE FROM script_transactions"
+     pure $ Just h
+
+
+-- For resuming we need to provide a list of points where we can resume from.
+
+instance Resumable ScriptTxHandle where
+  resumeFromStorage h = do
+    es <- Storable.getStoredEvents h
+    -- The ordering here matters. The node will try to find the first point in the
+    -- ledger, then move to the next and so on, so we will send the latest point
+    -- first.
+    pure $ map chainPoint es ++ [ChainPointAtGenesis]
 
 open :: FilePath -> Depth -> IO ScriptTxIndexer
 open dbPath (Depth k) = do
   c <- SQL.open dbPath
-  SQL.execute_ c "CREATE TABLE IF NOT EXISTS script_transactions (scriptAddress TEXT NOT NULL, txCbor BLOB NOT NULL, slotNo INT NOT NULL)"
+  SQL.execute_ c "CREATE TABLE IF NOT EXISTS script_transactions (scriptAddress TEXT NOT NULL, txCbor BLOB NOT NULL, slotNo INT NOT NULL, blockHash BLOB NOT NULL)"
```

Comment:
Do we need to create a Primary index, unique index, constraint for this table as well? 
We don't drop the table, we create it if it doesn't exist and insert into it. From SQL prospective there are no constraint that stops us from inserting the same row multiple times. Such Constraint will allow for `Insert or Replace` at SQL level

Reply:
I will add some indexes. Good catch!